js/ui/file_uploader.js

```diff
@@ -1597,6 +1363,237 @@ FileUploader.__internals = {
+            ajax.sendRequest({
+                url: this.fileUploader.option("uploadUrl"),
+                method: this.fileUploader.option("uploadMethod"),
+                headers: this.fileUploader.option("uploadHeaders"),
+                beforeSend: function(xhr) {
+                    file.request = xhr;
+                },
+                upload: {
+                    "onloadstart": function() {
+                        if(!file.isStartLoad) {
+                            file.isStartLoad = true;
+                            file.onLoadStart.fire();
+                        }
+                    },
+                    "onabort": function() {
+                        file.onAbort.fire();
+                        chunksData.chunks = [];
+                    }
+                },
+                data: this._createFormData(chunksData.name, this.fileUploader.option("name"),
+                    chunk.blob,
+                    chunk.index,
+                    chunksData.count,
+                    chunksData.type,
+                    chunksData.guid,
+                    chunksData.fileSize)
+            }).done(function() {
+                file.onProgress.fire({
+                    loaded: chunksData.loadedBytes,
+                    total: file.value.size
+                });
+                if(chunksData.chunks.length === 0) {
+                    file.onLoad.fire();
+                }
+                this._sendChunk(file, chunksData);
+            }.bind(this)).fail(function(e) {
+                if(this._isStatusError(e.status)) {
+                    file._isError = true;
+                    file.onError.fire();
+                }
+                chunksData.chunks = [];
+            }.bind(this));
+        }
+    },
+    _createFormData: function(fileName, blobName, blob, index, count, type, guid, size) {
+        var formData = new window.FormData();
+        formData.append(blobName, blob);
+        formData.append(FILEUPLOADER_CHUNK_META_DATA_NAME, JSON.stringify({
+            Name: fileName,
+            Index: index,
+            Count: count,
+            FileSize: size,
+            Type: type,
+            Guid: guid
+        }));
+        return formData;
+    },
+    _createChunkArray: function(file) {
+        var blobPosition = 0;
+        var chunkIndex = 0;
+        var result = [];
+        while(blobPosition <= file.size) {
+            result.push({
+                blob: this._sliceFile(file, blobPosition, this.chunkSize),
+                index: chunkIndex
+            });
+            blobPosition += this.chunkSize;
+            chunkIndex++;
+        }
+        return result;
+    },
+    _sliceFile: function(file, startPos, length) {
+        if(file.slice) {
+            return file.slice(startPos, startPos + length);
+        }
+        if(file.webkitSlice) {
+            return file.webkitSlice(startPos, startPos + length);
+        }
+        return null;
+    },
+    _getEvent: function(e) {
```

Comment:
Why need the "_getEvent" method?

Reply:
Maybe to set "e" to "event" always inside the base strategy.

Reply:
In whole file uploading mode, to an onProgress event is passed event from the ajax request.
In the case of uploading by chunks, each chunk has own request, so it's useless to forward it to an onProgress event.

Reply:
Probably, to have a single style in the API, it makes sense to pass the request event